### ISSUE TITLE
freqai - timerange bug in dataprovider.py with long training period

### ISSUE THREAD

**Issue by hamzz91** [freqtrade/freqtrade#9432]
## Describe your environment

  * Operating system: windows
  * Python Version: 3.11
  * CCXT version: 4.1.8
  * Freqtrade Version: freqtrade 2023.10-dev-cfda8516e
  

## Describe the problem:

I use a freqai strategy with 5m timeframe with "train_period_days"= 180 . If I put "1d" in include_timeframes, the lib fails

### Steps to reproduce:

  1. downlad the data : `freqtrade download-data -c .\config_examples\config_freqai.example_modif.json --timeframes 5m 1d --days 1800`
  2. Use the FreqaiExampleStrategy from the template and the attached config file [config_freqai.example_modif.json](https://github.com/freqtrade/freqtrade/files/13394210/config_freqai.example_modif.json) and any model (catboostRegressor for example)
  3. run the bot (I used the webserver, not sure if it's related)
  
### Observed Results:

  Using the debugger, I could see the "timerange" object of the dataprovider gets corrupted (the properties "startfmt" and "startdt" throw an error). This happens after the line 318 in the dataprovider.py file : `timerange.subtract_start(tf_seconds * startup_candles)`
The "startts" variable becomes negativ.

### Relevant code exceptions or logs

Note: Please copy/paste text of the messages, no screenshots of logs please.

  ```
2023-11-17 16:59:06,997 - freqtrade.rpc.api_server.api_backtest - ERROR - Backtesting caused an error: [Errno 22] Invalid argument
Traceback (most recent call last):
  File "E:\PycharmProjects\cryptobot\freqtrade\rpc\api_server\api_backtest.py", line 84, in __run_backtest_bg
    min_date, max_date = ApiBG.bt['bt'].backtest_one_strategy(
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\PycharmProjects\cryptobot\freqtrade\optimize\backtesting.py", line 1291, in backtest_one_strategy
    preprocessed = self.strategy.advise_all_indicators(data)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\PycharmProjects\cryptobot\freqtrade\strategy\interface.py", line 1335, in advise_all_indicators
    return {pair: self.advise_indicators(pair_data.copy(), {'pair': pair}).copy()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\PycharmProjects\cryptobot\freqtrade\strategy\interface.py", line 1335, in <dictcomp>
    return {pair: self.advise_indicators(pair_data.copy(), {'pair': pair}).copy()
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\PycharmProjects\cryptobot\freqtrade\strategy\interface.py", line 1367, in advise_indicators
    return self.populate_indicators(dataframe, metadata)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\PycharmProjects\cryptobot\user_data\strategies\FreqaiExampleStrategy.py", line 239, in populate_indicators
    dataframe = self.freqai.start(dataframe, metadata, self)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\PycharmProjects\cryptobot\freqtrade\freqai\freqai_interface.py", line 161, in start
    dk = self.start_backtesting(dataframe, metadata, self.dk, strategy)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\PycharmProjects\cryptobot\freqtrade\freqai\freqai_interface.py", line 323, in start_backtesting
    dataframe = self.dk.use_strategy_to_populate_indicators(
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\PycharmProjects\cryptobot\freqtrade\freqai\data_kitchen.py", line 798, in use_strategy_to_populate_indicators
    dataframe = self.populate_features(dataframe.copy(), pair, strategy,
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\PycharmProjects\cryptobot\freqtrade\freqai\data_kitchen.py", line 710, in populate_features
    informative_df = self.get_pair_data_for_features(
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\PycharmProjects\cryptobot\freqtrade\freqai\data_kitchen.py", line 671, in get_pair_data_for_features
    dataframe = strategy.dp.get_pair_dataframe(pair=pair, timeframe=tf)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\PycharmProjects\cryptobot\freqtrade\data\dataprovider.py", line 369, in get_pair_dataframe
    data = self.historic_ohlcv(pair=pair, timeframe=timeframe, candle_type=candle_type)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\PycharmProjects\cryptobot\freqtrade\data\dataprovider.py", line 319, in historic_ohlcv
    self.__cached_pairs_backtesting[saved_pair] = load_pair_history(
                                                  ^^^^^^^^^^^^^^^^^^
  File "E:\PycharmProjects\cryptobot\freqtrade\data\history\history_utils.py", line 57, in load_pair_history
    return data_handler.ohlcv_load(pair=pair,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\PycharmProjects\cryptobot\freqtrade\data\history\idatahandler.py", line 331, in ohlcv_load
  File "E:\PycharmProjects\cryptobot\freqtrade\data\history\idatahandler.py", line 383, in _validate_pairdata
    if pairdata.iloc[0]['date'] > timerange.startdt:
                                  ^^^^^^^^^^^^^^^^^
  File "E:\PycharmProjects\cryptobot\freqtrade\configuration\timerange.py", line 36, in startdt
    return datetime.fromtimestamp(self.startts, tz=timezone.utc)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
OSError: [Errno 22] Invalid argument

  ```

**Comment by robcaulk** [comment]
Hello,

Although we reminded you in the discord to provide full logs, they still appear to be missing. Please attach the full logs.

You also say “you don’t know if it is because of the webserver or not.” Can you please instead, run standard benchmark backtesting according to the FreqAI documentation, which means using CLI? We provided you this link in the discord, but it appears you may have ignored it. Thank you for this time following the link and using the instructions contained within:

https://www.freqtrade.io/en/latest/freqai-running/#backtesting

thank you,

Rob

**Comment by hamzz91** [comment]
Hello Rob,

Sorry for not being attentive. I tried with the regular backtesting feature with the same result :

`(.env) PS E:\PycharmProjects\cryptobot> freqtrade backtesting -s FreqaiExampleStrategy -c .\config_examples\config_freqai.example_modif.json --freqaimodel CatboostRegressor --timerange 20210101-20231023 --timeframe 5m
`

> 2023-11-18 11:26:30,333 - freqtrade - INFO - freqtrade 2023.10-dev-cfda8516e
> 2023-11-18 11:26:30,333 - freqtrade.configuration.load_config - INFO - Using config: .\config_examples\config_freqai.example_modif.json ...
> 2023-11-18 11:26:30,333 - freqtrade.loggers - INFO - Verbosity set to 0                                                                    
> 2023-11-18 11:26:30,333 - freqtrade.configuration.configuration - WARNING - `force_entry_enable` RPC message enabled.                      
> 2023-11-18 11:26:30,333 - freqtrade.configuration.configuration - INFO - Parameter -i/--timeframe detected ... Using timeframe: 5m ...     
> 2023-11-18 11:26:30,333 - freqtrade.configuration.configuration - INFO - Using max_open_trades: 5 ...                                      
> 2023-11-18 11:26:30,333 - freqtrade.configuration.configuration - INFO - Parameter --timerange detected: 20210101-20231023 ...             
> 2023-11-18 11:26:30,333 - freqtrade.configuration.configuration - INFO - Using user-data directory: E:\PycharmProjects\cryptobot\user_data ...
> 2023-11-18 11:26:30,333 - freqtrade.configuration.configuration - INFO - Using data directory: E:\PycharmProjects\cryptobot\user_data\data\binance ...
> 2023-11-18 11:26:30,333 - freqtrade.configuration.configuration - INFO - Overriding timeframe with Command line argument
> 2023-11-18 11:26:30,333 - freqtrade.configuration.configuration - INFO - Parameter --cache=day detected ...
> 2023-11-18 11:26:30,333 - freqtrade.configuration.configuration - INFO - Filter trades by timerange: 20210101-20231023
> 2023-11-18 11:26:30,333 - freqtrade.configuration.configuration - INFO - Using freqaimodel class name: CatboostRegressor
> 2023-11-18 11:26:30,333 - freqtrade.exchange.check_exchange - INFO - Checking exchange...
> 2023-11-18 11:26:30,349 - freqtrade.exchange.check_exchange - INFO - Exchange "binance" is officially supported by the Freqtrade development team.
> 2023-11-18 11:26:30,349 - freqtrade.configuration.configuration - INFO - Using pairlist from configuration.
> 2023-11-18 11:26:30,349 - freqtrade.configuration.config_validation - INFO - Validating configuration ...
> 2023-11-18 11:26:30,349 - freqtrade.commands.optimize_commands - INFO - Starting freqtrade in Backtesting mode
> 2023-11-18 11:26:30,349 - freqtrade.exchange.exchange - INFO - Instance is running with dry_run enabled
> 2023-11-18 11:26:30,349 - freqtrade.exchange.exchange - INFO - Using CCXT 4.1.8
> 2023-11-18 11:26:30,349 - freqtrade.exchange.exchange - INFO - Using Exchange "Binance"
> 2023-11-18 11:26:35,449 - freqtrade.resolvers.exchange_resolver - INFO - Using resolved exchange 'Binance'...
> 2023-11-18 11:26:35,668 - freqtrade.resolvers.iresolver - INFO - Using resolved strategy FreqaiExampleStrategy from 'E:\PycharmProjects\cryptobot\user_data\strategies\FreqaiExampleStrategy.py'...
> 2023-11-18 11:26:35,668 - freqtrade.strategy.hyper - INFO - Found no parameter file.
> 2023-11-18 11:26:35,668 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'timeframe' with value in config file: 5m.
> 2023-11-18 11:26:35,668 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'stake_currency' with value in config file: USDT.
> 2023-11-18 11:26:35,668 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'stake_amount' with value in config file: 200.
> 2023-11-18 11:26:35,668 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'unfilledtimeout' with value in config file: {'entry': 10, 'exit': 30, 'exit_timeout_count': 0, 'unit': 'minutes'}.
> 2023-11-18 11:26:35,668 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'max_open_trades' with value in config file: 5.
> 2023-11-18 11:26:35,668 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using minimal_roi: {'0': 0.1, '240': -1}
> 2023-11-18 11:26:35,668 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using timeframe: 5m
> 2023-11-18 11:26:35,668 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using stoploss: -0.05
> 2023-11-18 11:26:35,668 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_stop: False
> 2023-11-18 11:26:35,668 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_stop_positive_offset: 0.0
> 2023-11-18 11:26:35,668 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_only_offset_is_reached: False
> 2023-11-18 11:26:35,668 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using use_custom_stoploss: False
> 2023-11-18 11:26:35,668 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using process_only_new_candles: True
> 2023-11-18 11:26:35,668 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using order_types: {'entry': 'limit', 'exit': 'limit', 'stoploss': 'limit', 'stoploss_on_exchange': False, 'stoploss_on_exchange_interval': 60}   
> 2023-11-18 11:26:35,668 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using order_time_in_force: {'entry': 'GTC', 'exit': 'GTC'}
> 2023-11-18 11:26:35,668 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using stake_currency: USDT
> 2023-11-18 11:26:35,668 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using stake_amount: 200
> 2023-11-18 11:26:35,668 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using protections: []
> 2023-11-18 11:26:35,668 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using startup_candle_count: 40
> 2023-11-18 11:26:35,668 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using unfilledtimeout: {'entry': 10, 'exit': 30, 'exit_timeout_count': 0, 'unit': 'minutes'}
> 2023-11-18 11:26:35,668 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using use_exit_signal: True
> 2023-11-18 11:26:35,668 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using exit_profit_only: False
> 2023-11-18 11:26:35,668 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using ignore_roi_if_entry_signal: False
> 2023-11-18 11:26:35,668 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using exit_profit_offset: 0.0
> 2023-11-18 11:26:35,668 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using disable_dataframe_checks: False
> 2023-11-18 11:26:35,668 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using ignore_buying_expired_candle_after: 0
> 2023-11-18 11:26:35,668 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using position_adjustment_enable: False
> 2023-11-18 11:26:35,668 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using max_entry_position_adjustment: -1
> 2023-11-18 11:26:35,668 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using max_open_trades: 5
> 2023-11-18 11:26:35,668 - freqtrade.configuration.config_validation - INFO - Validating configuration ...
> 2023-11-18 11:26:35,683 - freqtrade.resolvers.iresolver - INFO - Using resolved pairlist StaticPairList from 'E:\PycharmProjects\cryptobot\freqtrade\plugins\pairlist\StaticPairList.py'...
> 2023-11-18 11:26:35,683 - freqtrade.data.dataprovider - INFO - Increasing startup_candle_count for freqai to 51880
> 2023-11-18 11:26:35,683 - freqtrade.data.history.history_utils - INFO - Using indicator startup period: 51880 ...
> 2023-11-18 11:26:35,868 - freqtrade.optimize.backtesting - INFO - Loading data from 2020-07-04 20:40:00 up to 2023-10-23 00:00:00 (1205 days).
> 2023-11-18 11:26:35,868 - freqtrade.optimize.backtesting - INFO - Dataload complete. Calculating indicators
> 2023-11-18 11:26:35,868 - freqtrade.optimize.backtesting - INFO - Running backtesting for Strategy FreqaiExampleStrategy
> 2023-11-18 11:26:36,015 - freqtrade.resolvers.iresolver - INFO - Using resolved freqaimodel CatboostRegressor from 'E:\PycharmProjects\cryptobot\freqtrade\freqai\prediction_models\CatboostRegressor.py'...
> 2023-11-18 11:26:36,015 - freqtrade.freqai.freqai_interface - INFO - Backtesting module configured to save all models.
> 2023-11-18 11:26:36,015 - freqtrade.freqai.data_drawer - INFO - Could not find existing datadrawer, starting from scratch
> 2023-11-18 11:26:36,015 - freqtrade.freqai.data_drawer - INFO - Could not find existing historic_predictions, starting from scratch
> 2023-11-18 11:26:36,015 - freqtrade.freqai.freqai_interface - INFO - Set fresh train queue from whitelist. Queue: ['BTC/USDT']
> 2023-11-18 11:26:36,015 - freqtrade.strategy.hyper - INFO - No params for buy found, using default values.
> 2023-11-18 11:26:36,031 - freqtrade.strategy.hyper - INFO - Strategy Parameter(default): std_dev_multiplier_buy = 1.25
> 2023-11-18 11:26:36,031 - freqtrade.strategy.hyper - INFO - No params for sell found, using default values.
> 2023-11-18 11:26:36,031 - freqtrade.strategy.hyper - INFO - Strategy Parameter(default): std_dev_multiplier_sell = 1.25
> 2023-11-18 11:26:36,031 - freqtrade.strategy.hyper - INFO - No params for protection found, using default values.
> 2023-11-18 11:26:36,031 - freqtrade.freqai.freqai_interface - INFO - Training 35 timeranges
> 2023-11-18 11:26:36,031 - freqtrade.freqai.freqai_interface - INFO - Training BTC/USDT, 1/1 pairs from 2020-07-05 00:00:00 to 2021-01-01 00:00:00, 1/35 trains
> 2023-11-18 11:26:36,031 - freqtrade.freqai.data_kitchen - INFO - Could not find backtesting prediction file at E:\PycharmProjects\cryptobot\user_data\models\uniqe-id\backtesting_predictions\cb_btc_1609459200_prediction.feather  
> 2023-11-18 11:26:36,046 - freqtrade.data.dataprovider - INFO - Increasing startup_candle_count for freqai to 51880
> 2023-11-18 11:26:37,039 - freqtrade.data.dataprovider - INFO - Increasing startup_candle_count for freqai to 51880
> 2023-11-18 11:26:37,047 - freqtrade - ERROR - Fatal exception!
> Traceback (most recent call last):
>   File "E:\PycharmProjects\cryptobot\freqtrade\main.py", line 42, in main
>     return_code = args['func'](args)
>                   ^^^^^^^^^^^^^^^^^^
>   File "E:\PycharmProjects\cryptobot\freqtrade\commands\optimize_commands.py", line 58, in start_backtesting
>     backtesting.start()
>   File "E:\PycharmProjects\cryptobot\freqtrade\optimize\backtesting.py", line 1374, in start
>     min_date, max_date = self.backtest_one_strategy(strat, data, timerange)
>                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "E:\PycharmProjects\cryptobot\freqtrade\optimize\backtesting.py", line 1291, in backtest_one_strategy
>     preprocessed = self.strategy.advise_all_indicators(data)
>                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "E:\PycharmProjects\cryptobot\freqtrade\strategy\interface.py", line 1335, in advise_all_indicators
>     return {pair: self.advise_indicators(pair_data.copy(), {'pair': pair}).copy()
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "E:\PycharmProjects\cryptobot\freqtrade\strategy\interface.py", line 1335, in <dictcomp>
>     return {pair: self.advise_indicators(pair_data.copy(), {'pair': pair}).copy()
>                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "E:\PycharmProjects\cryptobot\freqtrade\strategy\interface.py", line 1367, in advise_indicators
>     return self.populate_indicators(dataframe, metadata)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "E:\PycharmProjects\cryptobot\user_data\strategies\FreqaiExampleStrategy.py", line 239, in populate_indicators
>     dataframe = self.freqai.start(dataframe, metadata, self)
>                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "E:\PycharmProjects\cryptobot\freqtrade\freqai\freqai_interface.py", line 161, in start
>     dk = self.start_backtesting(dataframe, metadata, self.dk, strategy)
>          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "E:\PycharmProjects\cryptobot\freqtrade\freqai\freqai_interface.py", line 323, in start_backtesting
>     dataframe = self.dk.use_strategy_to_populate_indicators(
>                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "E:\PycharmProjects\cryptobot\freqtrade\freqai\data_kitchen.py", line 798, in use_strategy_to_populate_indicators
>     dataframe = self.populate_features(dataframe.copy(), pair, strategy,
>                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "E:\PycharmProjects\cryptobot\freqtrade\freqai\data_kitchen.py", line 710, in populate_features
>     informative_df = self.get_pair_data_for_features(
>                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "E:\PycharmProjects\cryptobot\freqtrade\freqai\data_kitchen.py", line 671, in get_pair_data_for_features
>     dataframe = strategy.dp.get_pair_dataframe(pair=pair, timeframe=tf)
>                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "E:\PycharmProjects\cryptobot\freqtrade\data\dataprovider.py", line 369, in get_pair_dataframe
>     data = self.historic_ohlcv(pair=pair, timeframe=timeframe, candle_type=candle_type)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "E:\PycharmProjects\cryptobot\freqtrade\data\dataprovider.py", line 319, in historic_ohlcv
>     self.__cached_pairs_backtesting[saved_pair] = load_pair_history(
>                                                   ^^^^^^^^^^^^^^^^^^
>   File "E:\PycharmProjects\cryptobot\freqtrade\data\history\history_utils.py", line 57, in load_pair_history
>     return data_handler.ohlcv_load(pair=pair,
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "E:\PycharmProjects\cryptobot\freqtrade\data\history\idatahandler.py", line 331, in ohlcv_load
>     self._validate_pairdata(pair, pairdf, timeframe, candle_type, timerange_startup)
>   File "E:\PycharmProjects\cryptobot\freqtrade\data\history\idatahandler.py", line 383, in _validate_pairdata
>     if pairdata.iloc[0]['date'] > timerange.startdt:
>                                   ^^^^^^^^^^^^^^^^^
>   File "E:\PycharmProjects\cryptobot\freqtrade\configuration\timerange.py", line 36, in startdt
>     return datetime.fromtimestamp(self.startts, tz=timezone.utc)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> OSError: [Errno 22] Invalid argument

**Comment by xmatthias** [comment]
> Increasing startup_candle_count for freqai to 51880

Well - if you do the math, then on a daily timeframe, that's 142 years - which will bring you (well) before 1970 (which would be epoch==0) - and which is what the error is about.
Funnily, if you'd not be running windows, this would not fail (though it'd still not work for different reasons in freqAI).

![image](https://github.com/freqtrade/freqtrade/assets/5024695/546c351c-3dce-441e-8e5b-e4ea4917a75e)


effectively - with this configuration (amount of startup candles, in combination with the "daily" timeframe), you'll not be able to successfully run backtesting (nor live, most likely).
While it may not crash (we can fix that part by limiting startts to be positive)  - it'll still not work, as you'll have too many NaN values (no data much prior to 2017, usually), which freqAI automatically drops.

**Comment by robcaulk** [comment]
Does Bybit actually have futures data on your pairs back to 2021? When I use the example config, it tells me 1INCH has data starting at 2023-05-01. But as long as I backtest pairs within the timerange that the exchange actually provides, everything works perfectly.

Also, please do not modify the startup_candle_count in your strategy.

**Comment by robcaulk** [comment]
Hello, please test this branch and tell us if your problem is resolved:

https://github.com/freqtrade/freqtrade/pull/9450

Thank you,

Rob

**Comment by hamzz91** [comment]
Hi Rob,

Sure, I will test it and let you know.

Thank you for your help.

Sincerely

**Comment by hamzz91** [comment]
Hi Rob,

I tested the fix and the issue does not appear anymore if I add "1d" in the "include_timeframes" config file. However, an exception appeared on the model training part : the dataset was empty. It happened because all rows containing a NaN when building the train/test datasets were deleted. So there was some columns containing only NaN values. After checking the dates, I could see the `informative_df` used to compute the indicators (line 710 of **data_kitchen.py**, function **populate_features**) was not including the 180 training days. I fixed this by modifying the `add_train_candles` parameter to True in the **dataprovider.py** file, line 316 (funciton **historic_ohlcv**). Not sure if it's the correct fix, but it worked fine on my situation.

Thank you !

**Comment by xmatthias** [comment]
i think this was fixed in #9450.

**Comment by robcaulk** [comment]
> Hi Rob,
> 
> I tested the fix and the issue does not appear anymore if I add "1d" in the "include_timeframes" config file. However, an exception appeared on the model training part : the dataset was empty. It happened because all rows containing a NaN when building the train/test datasets were deleted. So there was some columns containing only NaN values. After checking the dates, I could see the `informative_df` used to compute the indicators (line 710 of **data_kitchen.py**, function **populate_features**) was not including the 180 training days. I fixed this by modifying the `add_train_candles` parameter to True in the **dataprovider.py** file, line 316 (funciton **historic_ohlcv**). Not sure if it's the correct fix, but it worked fine on my situation.
> 
> Thank you !

Can you please push a PR for this so we can see more clearly what you are doing?
